frontend/src/components/samples/SampleEditContent.js

```diff
@@ -158,14 +155,30 @@ const SampleEditContent = ({ token, samplesByID, sampleKinds, add, update, listT
     onSearchContainer(newData.container, { exact_match: true })
     onSearchCoordinate(newData.coordinate, { exact_match: true })
     onSearchSampleKind(newData.sample_kind)
+    if (!isAdding)
+      form.setFieldsValue({ ...newData })
   }, [sampleValue])
 
+  const sampleKind = (sampleKindID) => sampleKinds.itemsByID[sampleKindID]
+
   const onValuesChange = (values) => {
-    setFormData(deserialize({ ...formData, ...values }))
+    if (values["sample_kind"]) {
+      if (!sampleKind(values["sample_kind"]).is_extracted) {
+        form.setFieldValue('tissue_source', '')
+        setisTissueEnabled(false)
+      } else {
+        setisTissueEnabled(true)
+      }
+    }
+    if (values["container"]) {
+      setIsCoordRequired(true)
+    } else {
+      setIsCoordRequired(false)
```

Comment:
The coordinates field gets disabled here every time a value other than 'container' changes.